### PR TITLE
summary function updated to correctly extract the full alpha level + test for this added

### PR DESCRIPTION
**Pull request by yuliasidi**

### CODE REVIEW

**Comment by yuliasidi** [comment]
@nanxstats, @LittleBeannie, the check is failing but not due to the PR. Please advise

File: R/summary.R [hunk 1]
```diff
@@ -692,10 +692,10 @@ summary.gs_design <- function(object,
   # Save the full alpha as an attribute of the output summary table
   attr(output, "full_alpha") <-
     if (is.null(object$input$alpha)) {
-      if (is.null(object$upar$total_spend)) {
+      if (is.null(object$input$upar$total_spend)) {
         0.025
       } else {
-        object$upar$total_spend
+        object$input$upar$total_spend
```
Comment (jdblischak) [inline R/summary.R:698]: This line is causing the vignette `story-nph-futility.Rmd` to fail. `upar` is a numeric vector, not a named list

```R
wieand <- gs_power_ahr(
  enroll_rate = enroll_rate, fail_rate = fail_rate,
  upper = gs_b, upar = c(rep(Inf, 2), qnorm(.975)),
  lower = gs_b, lpar = c(0, 0, -Inf),
  event = 512 * c(.5, .75, 1)
)

str(wieand$input$upar)
##  num [1:3] Inf Inf 1.96
```

@LittleBeannie could you please investigate? Do we need to update to give `upar` names, or do we need to change how the total spend is extracted?

https://github.com/Merck/gsDesign2/blob/ee7806593e40e0887dec5d0b899500d69feff35c/vignettes/articles/story-nph-futility.Rmd#L89-L96

File: tests/testthat/test-developer-summary.R [hunk 2]
```diff
@@ -149,3 +149,21 @@ test_that("The column 'Bound' is always included in summary.gs_design() output",
...
+      total_spend = a_level
+    ),
+    test_lower = FALSE
+  )
+
+  # without col_vars
+  observed <- summary(x)
+
+  expect_equal(attributes(observed)$full_alpha, a_level)
+})
```
Comment (jdblischak) [inline tests/testthat/test-developer-summary.R:169]: Nice test!

**Review by jdblischak — COMMENTED** [review]
Thanks for the PR! Unfortunately the pkgdown [error](https://github.com/Merck/gsDesign2/actions/runs/9648335533/job/26609255201?pr=426#step:6:152) is caused by this PR. Previously this problematic code was never executed because without `$input`, it always evaluated to `NULL`.

```R
✖ Failed to render RMarkdown document.
  
  Quitting from lines 89-101 [unnamed-chunk-4] (story-nph-futility.Rmd)
Caused by error:
! $ operator is invalid for atomic vectors
Backtrace:
    ▆
 1. ├─wieand %>% summary() %>% ...
 2. ├─gsDesign2::as_gt(...)
 3. ├─base::summary(.)
 4. └─gsDesign2:::summary.gs_design(.)
```

File: R/summary.R (hunk 1 above)
Comment (jdblischak) [reply R/summary.R:698]: @LittleBeannie The code below reproduces the error locally for me. The message matches the error in the failing pkgdown job. Could you please run it locally to see if you can also reproduce the error?

```R
remotes::install_github("Merck/gsDesign2@summary_upar", upgrade = FALSE)
library("gsDesign2")
enroll_rate <- define_enroll_rate(duration = 12, rate = 680 / 12)
fail_rate <- define_fail_rate(
  duration = c(3, 100),
  fail_rate = -log(.5) / 12,
  hr = c(1, .693),
  dropout_rate = 0
)
wieand <- gs_power_ahr(
  enroll_rate = enroll_rate, fail_rate = fail_rate,
  upper = gs_b, upar = c(rep(Inf, 2), qnorm(.975)),
  lower = gs_b, lpar = c(0, 0, -Inf),
  event = 512 * c(.5, .75, 1)
)
summary(wieand)
## Error in object$input$upar$total_spend :
##   $ operator is invalid for atomic vectors
```

Comment (LittleBeannie) [reply R/summary.R:698]: Thanks for the confirmation, @jdblischak . I tried it again and it runs without error.
image

Comment (jdblischak) [reply R/summary.R:698]: @LittleBeannie please run the following. Does your code have the `input` between `object` and `upar`?

```R
grep("total_spend", deparse(gsDesign2:::summary.gs_design), value = TRUE)
## [1] "        if (is.null(object$input$upar$total_spend)) {"
## [2] "            object$input$upar$total_spend"  
```

Comment (LittleBeannie) [reply R/summary.R:698]: Here is my output:
image

Comment (jdblischak) [reply R/summary.R:698]: You aren't using the code in this PR. @yuliasidi updated the lines to be `object$input$upar$total_spend`. You won't observe the error unless you install the code from this PR

```R
remotes::install_github("Merck/gsDesign2@summary_upar", upgrade = FALSE)
```

Comment (LittleBeannie) [reply R/summary.R:699]: My bad... I installed Yulia's branch and run the above codes locally, and get the following error. I have get it fixed in my latest commit.
image

**Review by LittleBeannie — APPROVED** [review]
Thanks, @yuliasidi for fixing the bug!

**Review by jdblischak — APPROVED** [review]

**Pull request merged**